api/tests/routes/pro/patch_offerer_bank_accounts_test.py

```diff
@@ -0,0 +1,479 @@
+            assert action_logged.bankAccountId == action_occured.bankAccountId
+
+    @pytest.mark.usefixtures("db_session")
+    def test_user_cannot_link_venue_to_a_bank_account_that_doesnt_depend_on_its_offerer(self, db_session, client):
+        actions_occured = []
+
+        offerer = offerers_factories.OffererFactory()
+        another_offerer = offerers_factories.OffererFactory()
+        pro_user = users_factories.ProFactory()
+        offerers_factories.UserOffererFactory(user=pro_user, offerer=offerer)
+        bank_account = finance_factories.BankAccountFactory(offerer=offerer)
+        bank_account_of_another_offerer = finance_factories.BankAccountFactory(offerer=another_offerer)
+        bank_account_of_another_offerer_id = bank_account_of_another_offerer.id
+        venue = offerers_factories.VenueFactory(managingOfferer=offerer)
+
+        assert not bank_account.venueLinks
+
+        http_client = client.with_session_auth(pro_user.email)
+
+        # Fetch user session
+        # Fetch user
+        # Check user permission
+        # Fetch bank account, links and offerer
+        with testing.assert_num_queries(4):
+            response = http_client.patch(
+                f"/offerers/{offerer.id}/bank-accounts/{bank_account_of_another_offerer_id}",
+                json={"venues_ids": [venue.id]},
+            )
+
+        assert response.status_code == 404
+
+        response = http_client.get(f"/offerers/{offerer.id}/bank-accounts/")
+
+        assert response.status_code == 200
+        assert len(response.json["bankAccounts"]) == 1
+        bank_account_response = response.json["bankAccounts"].pop()
+        assert not bank_account_response["linkedVenues"]
+
+        db_session.refresh(bank_account)
+
+        assert not bank_account.venueLinks
+
+        actions_logged = history_models.ActionHistory.query.order_by(
+            history_models.ActionHistory.actionDate, history_models.ActionHistory.venueId
+        ).all()
+
+        assert len(actions_logged) == len(actions_occured) == 0
+
+    def test_user_cannot_link_venue_that_doesnt_depend_on_its_offerer_to_a_bank_account(self, db_session, client):
+        actions_occured = []
+
+        offerer = offerers_factories.OffererFactory()
+        another_offerer = offerers_factories.OffererFactory()
+        pro_user = users_factories.ProFactory()
+        offerers_factories.UserOffererFactory(user=pro_user, offerer=offerer)
+        bank_account = finance_factories.BankAccountFactory(offerer=offerer)
+        bank_account_id = bank_account.id
+        offerers_factories.VenueFactory(managingOfferer=offerer)
+        venue_of_another_offerer = offerers_factories.VenueFactory(managingOfferer=another_offerer)
+        venue_of_another_offerer_id = venue_of_another_offerer.id
+
+        assert not bank_account.venueLinks
+
+        http_client = client.with_session_auth(pro_user.email)
+
+        with testing.assert_no_duplicated_queries():
```

Comment:
Détail : `assert_no_duplicated_queries` est surtout efficace lorsqu'on a des listes, qu'on itère dans des boucles, il ne détectera pas forcément grand chose lorsqu'on ne traite qu'un seul objet.

Reply:
oopsi, ça n’avait pas vocation à rester